### PR TITLE
[ARTS-685] - Site service client implementation

### PR DESCRIPTION
**Pull request by anatbal**
## Description
- Initial implementation of site service client
- Minor changes in role-service-client to simplify the constructor (managed to make the configuration static)
- Tested on AZ
- Tested on local profile

### Changes
- [ARTS-685 ] -implementation of site service client

### Checklist:

- [x] Branch name follows convention (feature/arts-#-short-name OR fix/arts-#-short-name)
- [x] I have included a short-meaningful title, description and changes for this PR

### CODE REVIEW

File: site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/SiteServiceClient.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    public boolean entityIdExists(final String siteId,
+                                  final Consumer<HttpHeaders> authHeaders)
+            throws RestException {
+        Objects.requireNonNull(siteId, ResponseMessages.ID_NOT_NULL);
+        try {
+            getById(siteId, authHeaders);
+        } catch (RestException ex) {
+            return false;
+        }
+        return true;
```
Comment (tamirkamara) [inline site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/SiteServiceClient.java:58]: I think this logic can produce unwanted results - any RestExpection translates to not found entity isn't always the case...

File: site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/SiteServiceClient.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    public SiteDTO createEntity(final SiteDTO site,
+                                final Consumer<HttpHeaders> authHeaders) throws RestException {
```
Comment (tamirkamara) [inline site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/SiteServiceClient.java:62]: Does it throw RestException?

File: site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/SiteServiceClient.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    public List<SiteDTO> getAllSites(final Consumer<HttpHeaders> authHeaders) {
+        String uri = UriComponentsBuilder
+                .fromUriString(ClientRoutesConfig.getServiceGetAllSites())
+                .build().toString();
+        return Arrays.asList(requestExecutor.sendBlockingGetRequest(webClient, uri, SiteDTO[].class, authHeaders));
+    }
+
+    public SiteDTO getById
+            (final String siteId,
+             final Consumer<HttpHeaders> authHeaders) throws RestException {
```
Comment (tamirkamara) [inline site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/SiteServiceClient.java:78]: Does it throw RestException?

**Review by tamirkamara — CHANGES_REQUESTED** [review]

File: site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/SiteServiceClient.java (hunk 1 above)
Comment (anatbal) [reply site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/SiteServiceClient.java]: Fixed, but I'm not sure about it... The fix is not very webclient "native", tell me WDYT

File: site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/model/SiteAddress.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
+package uk.ac.ox.ndph.mts.siteserviceclient.model;
+
+/**
+ * Address Model - Common could be UK, US, encapsulates common address attributes
+ */
+public class SiteAddress {
```
[the remark this answers is not in the document]
Comment (anatbal) [reply site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/model/SiteAddress.java]: Fixed

File: site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/SiteServiceClient.java (hunk 2 above)
Comment (anatbal) [reply site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/SiteServiceClient.java]: Fixed
It's temp just an Exception like we deciced

File: site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/SiteServiceClient.java (hunk 3 above)
Comment (anatbal) [reply site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/SiteServiceClient.java]: Same

**Review by tamirkamara — DISMISSED** [review]
Have you checked that the changes work in "local" mode?

**Review by shajidissa — DISMISSED** [review]

File: site-service-client/src/test/java/uk/ac/ox/ndph/mts/siteserviceclient/common/MockWebServerWrapper.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
+package uk.ac.ox.ndph.mts.siteserviceclient.common;
```
Comment (eladiw) [inline site-service-client/src/test/java/uk/ac/ox/ndph/mts/siteserviceclient/common/MockWebServerWrapper.java:1]: should this mock be in this package (and not in the tests package)

File: site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/configuration/WebClientConfigSite.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+@Configuration
+@ConfigurationProperties(prefix = "http-client")
+public class WebClientConfigSite {
+
+    @Value("${http-client.connectTimeOutMs:3000}")
```
Comment (eladiw) [inline site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/configuration/WebClientConfigSite.java:24]: add todo to add the appropriate config file and add it to the config schema validator

**Review by eladiw — APPROVED** [review]

**Review by LizaShak — APPROVED** [review]

**Pull request merged**